src/plugins/console/public/lib/autocomplete/autocomplete.ts

```diff
@@ -349,14 +349,66 @@ export default function ({
     });
   }
 
+  function hasOneOf(value: unknown): value is { __one_of: ConditionalTemplateType[] } {
+    return typeof value === 'object' && value !== null && '__one_of' in value;
+  }
+
+  function getConditionalTemplate(
+    name: string,
+    autocompleteRules: Record<string, unknown> | null | undefined
+  ) {
+    if (autocompleteRules && autocompleteRules[name]) {
+      const currentLineNumber = editor.getCurrentPosition().lineNumber;
+      const value = autocompleteRules[name];
+
+      if (hasOneOf(value)) {
+        const startLine = getStartLineNumber(currentLineNumber, value.__one_of);
+        const lines = editor.getLines(startLine, currentLineNumber).join('\n');
+        const match = matchCondition(lines, value.__one_of);
+        if (match && match.__template) {
+          return match.__template;
+        }
+      }
+    }
+  }
+
+  function getStartLineNumber(currentLine: number, rules: ConditionalTemplateType[]): number {
+    if (currentLine === 1) {
+      return currentLine;
+    }
+    const value = editor.getLineValue(currentLine);
+    const match = matchCondition(value, rules);
+    if (match) {
+      return currentLine;
+    }
+    return getStartLineNumber(currentLine - 1, rules);
+  }
+
+  function matchCondition(condition: string, rules: ConditionalTemplateType[]) {
+    return rules.find((rule) => {
+      if (rule.__condition && rule.__condition.lines_regex) {
+        return new RegExp(rule.__condition.lines_regex, 'm').test(condition);
+      }
+      return false;
+    });
+  }
+
   function applyTerm(term: {
     value?: string;
     context?: AutoCompleteContext;
-    template?: { __raw: boolean; value: string };
+    template?: { __raw?: boolean; value?: string; [key: string]: unknown };
```

Comment:
Were the typings for this incorrect before?

Reply:
`template` can have optional `__raw` and `value` properties _"to determine whether this template should go through JSON formatting"_. See [here](https://github.com/elastic/kibana/blob/main/src/plugins/console/public/lib/autocomplete/autocomplete.ts#L380). In most cases, it is of type `[key: string]: unknown`.